x/pairing/keeper/unresponsive_provider.go

```diff
@@ -41,68 +42,65 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 		}
-		currentEpoch = previousEpoch
-	}
-
-	// Go over previous epochs to count the CU serviced by the provider
-	epochAfterCountingProviderCu, err := k.countCuForUnresponsiveProviderAndComplainers(ctx, true, currentEpoch, epochsNumToCheckCUForUnresponsiveProvider, providersPunishEntryMap)
-	if err != nil {
-		return nil, utils.LavaError(ctx, k.Logger(ctx), "count_cu_unresponsive_provider", map[string]string{"err": err.Error()}, "couldn't count the CU of the unresponsive provider")
-	}
-
-	// Go over previous epochs (from the point the last loop finished) to count the CU of clients that complained about the provider
-	_, err = k.countCuForUnresponsiveProviderAndComplainers(ctx, false, epochAfterCountingProviderCu, epochsNumToCheckCUForComplainers, providersPunishEntryMap)
-	if err != nil {
-		return nil, utils.LavaError(ctx, k.Logger(ctx), "count_cu_complainers", map[string]string{"err": err.Error()}, "couldn't count the CU of the complainers")
+		epochTemp = previousEpoch
 	}
 
-	// Get providers that should be punished
-	for punishEntryKey, punishEntry := range providersPunishEntryMap {
-		// provider doesn't have payments (serviced CU = 0)
-		if punishEntry.providerServicedCu == 0 {
-			chainID := strings.Split(punishEntryKey, "_")[1]
-
-			// Try getting the provider's stake entry
-			_, foundStakeEntry := k.epochStorageKeeper.GetEpochStakeEntries(ctx, currentEpoch, epochstoragetypes.ProviderKey, chainID)
-
-			// if stake entry is not found -> this is the provider's first epoch (didn't get a chance to stake yet) -> don't punish him
-			if !foundStakeEntry {
-				continue
+	// Go over the staked provider entries (on all chains)
+	for _, providerStakeStorage := range providerStakeStorageList {
+		for _, providerStakeEntry := range providerStakeStorage.GetStakeEntries() {
+			// update the CU count for this provider in providerCuCounterForUnreponsivenessMap
+			err := k.updateCuForProviderInProviderCuCounterForUnreponsivenessMap(ctx, epochTemp, epochsNumToCheckCUForUnresponsiveProvider, epochsNumToCheckCUForComplainers, providerStakeEntry, providerCuCounterForUnreponsivenessMap)
+			if err != nil {
+				return nil, utils.LavaError(ctx, k.Logger(ctx), "update_cu_for_provider_in_provider_cu_counter_for_unreponsiveness_map", map[string]string{"err": err.Error()}, "couldn't update the CU count in the providerCuCounterForUnreponsivenessMap")
 			}
-		}
 
-		// The provider served less CU than the sum of the complainers CU -> should be punished
-		if punishEntry.complainersUsedCu > punishEntry.providerServicedCu {
-			providersToPunishMap[punishEntryKey] = punishEntry.providerPaymentStorageIndexList
+			// get the provider's providerCuCounterForUnreponsiveness from the providerCuCounterForUnreponsivenessMap
+			providerCuCounterForUnreponsiveness, found, providerCuCounterForUnreponsivenessKey := k.getProviderCuCounterForUnreponsivenessEntryFromMap(providerStakeEntry, providerCuCounterForUnreponsivenessMap)
```

Comment:
why do we still need th map? return what you need from the function above and use it

Reply:
done